cmd/brew-test-bot.rb

```diff
@@ -552,18 +553,9 @@ def skip(formula_name)
     end
 
     def satisfied_requirements?(formula, spec, dependency = nil)
-      requirements = formula.send(spec).recursive_requirements
-
-      unsatisfied_requirements = requirements.reject do |requirement|
-        satisfied = false
-        satisfied ||= requirement.satisfied?
-        satisfied ||= requirement.optional?
-        if !satisfied && requirement.default_formula?
-          default = Formula[requirement.default_formula]
-          satisfied = satisfied_requirements?(default, :stable, formula.full_name)
-        end
-        satisfied
-      end
+      f = Formulary.factory(formula.full_name, spec)
+      fi = FormulaInstaller.new(f)
+      unsatisfied_requirements, _ = fi.expand_requirements
```

Comment:
You can skip the `_` here, by the way.

Feels a little weird using `FormulaInstaller` here. What do you (and @ilovezfs) think about maybe adding/moving the `expand_requirements` logic into `Formula`? I haven't looked at how gnarly that would be. If it's hard: this seems fine.

@ilovezfs any other thoughts here?